lahja/asyncio/endpoint.py

```diff
@@ -235,13 +245,12 @@ def can_send_item(self, item: BaseEvent, config: Optional[BroadcastConfig]) -> b
     async def send_message(self, message: Msg) -> None:
         await self.conn.send_message(message)
 
-    async def wait_until_subscription_received(self) -> None:
-        async with self._received_subscription:
-            await self._received_subscription.wait()
-
     async def wait_until_subscribed_to(self, event: Type[BaseEvent]) -> None:
-        while event not in self.subscribed_messages:
-            await self.wait_until_subscription_received()
+        async with self._received_subscription:
+            while True:
```

Comment:
@lithp can you confirm this is ok according to your understanding of conditions.  rather than release and then re-acquire at the entrance to each loop, we eliminate this and only switch on the release/acquire that happens as part of the `wait` call.

Reply:
This looks good to me!